### PR TITLE
feat: enable password changes in coder-sdk

### PR DESCRIPTION
**Pull request by jawnsy**
* Allow modification of passwords and the temporary_password flag for users managed by the built-in authentication provider.
* Split unit tests into separate files corresponding to the implementations.

### CODE REVIEW

File: coder-sdk/test/login_cmd.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+
+	"cdr.dev/coder-cli/coder-sdk"
+)
+
+func main() {
```
Comment (deansheather) [inline coder-sdk/test/login_cmd.go:11]: Why is this a main package if it's for testing?

File: coder-sdk/users_test.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+		HTTPClient: server.Client(),
+		Token:      "JcmErkJjju-KSrztst0IJX7xGJhKQPtfv",
+	})
+	require.NoError(t, err, "failed to create coder.Client")
+
+	err = client.UpdateUser(context.Background(), "me", coder.UpdateUserReq{
+		UserPasswordSettings: &coder.UserPasswordSettings{
+			OldPassword: "vt9g9rxsptrq",
+			Password:    "wmf39jw2f7pk",
+			Temporary:   true,
```
Comment (deansheather) [inline coder-sdk/users_test.go:100]: If you're changing your own password you're not intended to set temporary = true, the temporary flag is for when accounts are created or password is changed by an admin.

**Review by deansheather — APPROVED** [review]

File: coder-sdk/test/login_cmd.go (hunk 1 above)
Comment (jawnsy) [reply coder-sdk/test/login_cmd.go:11]: oops, did not mean to check this in

File: coder-sdk/users_test.go (hunk 2 above)
Comment (jawnsy) [reply coder-sdk/users_test.go]: fair enough, made this false

**Pull request merged**